jujugui/static/gui/src/app/components/deployment/summary/summary.js

```diff
@@ -0,0 +1,193 @@
+      // The env is already bound to ecsCommit in app.js.
+      this.props.ecsCommit();
+      this.setState({hasCommits: true}, () => {
+        this._close();
+      });
+    },
+
+    /**
+      Handle navigating to the machine view.
+
+      @method _handleViewMachinesClick
+    */
+    _handleViewMachinesClick: function() {
+      this.props.changeState({
+        sectionB: {
+          component: 'machine',
+          metadata: {}
+        },
+        sectionC: {
+          component: null,
+          metadata: {}
+        }
+      });
+    },
+
+    /**
+      Handle changes to the placement radio buttons.
+
+      @method _handlePlacementChange
+      @param {Object} e The click event.
+    */
+    _handlePlacementChange: function(e) {
+      this.setState({
+        autoPlace: e.currentTarget.getAttribute('data-placement') === 'placed'
+      });
+    },
+
+    /**
+      Generate the list of change items.
+
+      @method _generateChangeItems
+      @returns {Array} The collection of changes.
+    */
+    _generateChangeItems: function() {
+      var changeList = this.props.changeDescriptions;
+      var changes = [];
+      changeList.forEach(function(change, i) {
+        changes.push(
+          <juju.components.DeploymentSummaryChangeItem
+            key={i}
+            change={change} />
+          );
+      }, this);
+      return changes;
+    },
+
+    render: function() {
+      var listHeaderClassName = 'deployment-summary-change-item ' +
+          'deployment-summary__list-header';
+      var buttons = [{
+        title: 'Clear changes',
+        action: this._handleClear
+      }, {
+        title: 'Deploy',
+        action: this._handleDeploy,
+        type: 'confirm'
+      }];
+      return (
+        <div className="deployment-panel__child">
+          <div className="deployment-panel__content">
+            <div className="twelve-col">
+              <div className="inner-wrapper">
+                <h2 className="deployment-panel__title">
+                  Deployment summary
+                </h2>
+                <juju.components.DeploymentSummaryPlacement
+                  handleViewMachinesClick={this._handleViewMachinesClick}
+                  handlePlacementChange={this._handlePlacementChange}
+                  autoPlace={this.state.autoPlace}
+                  getUnplacedUnitCount={this.props.getUnplacedUnitCount} />
+                <ul className="deployment-summary__list">
+                  <li className={listHeaderClassName}>
+                    <span className="deployment-summary-change-item__change">
+                      Change
+                    </span>
+                    <span className="deployment-summary-change-item__time">
+                      Time
+                    </span>
+                  </li>
+                  {this._generateChangeItems()}
+                </ul>
+              </div>
+            </div>
+          </div>
+          <div className="deployment-panel__footer">
```

Comment:
So this is the pattern we'll need to follow on each one of these steps? Does it make sense to split this out into its own component for `DeploymentPanelFooter`?